azure_functions_worker/bindings/meta.py

```diff
@@ -25,15 +34,43 @@ def load_binding_registry() -> None:
     global BINDING_REGISTRY
     BINDING_REGISTRY = func.get_binding_registry()
 
-
-def get_binding(bind_name: str) -> object:
+    if BINDING_REGISTRY is None:
+        # If the BINDING_REGISTRY is None, azure-functions hasn't been
+        # loaded in properly.
+        raise AttributeError('BINDING_REGISTRY is None. Sys Path:'
+                             f'{sys.path}, Sys Module: {sys.modules},'
+                             f'python-packages Path exists: '
+                             f'{os.path.exists(CUSTOMER_PACKAGES_PATH)}.')
+
+    # The base extension supports python 3.8+
+    if sys.version_info.minor >= BASE_EXT_SUPPORTED_PY_MINOR_VERSION:
+        # Import the base extension
+        try:
+            import azurefunctions.extensions.base as clients
+            global DEFERRED_BINDING_REGISTRY
+            DEFERRED_BINDING_REGISTRY = clients.get_binding_registry()
+        except ImportError:
+            logger.info('Base extension not found. '
+                        f'Python version: 3.{sys.version_info.minor}, '
+                        f'Sys path: {sys.path}, '
+                        f'Sys Module: {sys.modules}, '
+                        f'python-packages Path exists: '
+                        f'{os.path.exists(CUSTOMER_PACKAGES_PATH)}.')
+
+
+def get_binding(bind_name: str,
```

Comment:
We should explore - for future - I do not like that we keep on doing `get_binding` in an inefficient way with each invocation call.

Reply:
Created future task for this.